### ISSUE TITLE
[bitnami/jaeger] Can't provide own secret for cassandra db via existingSecret

### ISSUE THREAD

**Issue by Sebastian-Materna** [bitnami/charts#26359]
### Name and Version

bitnami/jaeger 1.7,1

### What architecture are you using?

amd64

### What steps will reproduce the bug?

1. Use this Helm chart as a dependency
2. Create values
3. Install/Upgrade the Helm release

### Are you using any custom parameters or values?

We are using the Jaeger Helm chart as a dependency in our Helm chart. Part of our values.yaml:

```yaml
jaeger:
  externalDatabase:
    existingSecret: jaeger-secret
    existingSecretPasswordKey: cassandra-password
  cassandra:
    persistence:
      size: 8Gi
```

### What is the expected behavior?

I want to use an existing secret, which is created by external secrets operator named `jaeger-secret`. But this secret will not used, only the default `jaeger-cassandra`.

### What do you see instead?

A new secret `jaeger-cassandra` gets created.

**Comment by carrodher** [comment]
Are you able to reproduce the issue by deploying the Helm chart in a standalone way (without being a subchart)?

**Comment by Sebastian-Materna** [comment]
Yes, unfortunately:
Using the `values-test.yaml`:
```
externalDatabase:
  existingSecret: jaeger-secret
  existingSecretPasswordKey: cassandra-password

  cassandra:
    persistence:
      size: 24Gi
```
Running the command `helm template -n test -f values-test.yaml jaeger bitnami/jaeger --version 1.7.1 > test.yaml`
There are no matches for `jaeger-secret`, only the default from the template `jaeger-cassandra`. This happens also on the latest version (2.2.1).
```
            - name: CASSANDRA_PASSWORD
              valueFrom:
                secretKeyRef:
                  name: jaeger-cassandra
                  key: cassandra-password
```
I'm using this as a subchart, because I have added an Ingress resource and ExternalSecrets resource. Also the persistence can't be changed right now. I've set `24Gi`, but it's still `8Gi`

```
  volumeClaimTemplates:
    - apiVersion: v1
      kind: PersistentVolumeClaim
      metadata:
        name: data
        labels:
          app.kubernetes.io/instance: jaeger
          app.kubernetes.io/name: cassandra
      spec:
        accessModes:
          - "ReadWriteOnce"
        resources:
          requests:
            storage: "8Gi"
```

**Comment by migruiz4** [comment]
Hi @Sebastian-Materna,

The values `externalDatabase.existingSecret` and  `externalDatabase.existingSecretPasswordKey` are only taken into account when the Cassandra subchart is not used, which means `cassandra.enabled=false`.

To provide an existingSecret, you should use the value `cassandra.dbUser.existingSecret`. 
If using Jaeger as a subchart: `jaeger.cassandra.dbUser.existingSecret`.

**Comment by migruiz4** [comment]
While testing your use case, I found an issue where Jaeger would not use the `cassandra.dbUser.existingSecret` value, so you may need to update to the latest version once this fix is released: #26563
